Automattic/wp-calypso
# Update WordPress dependencies

Big cleanup of `@wordpress` dependencies in Calypso monorepo:

In `full-site-editing` and `wpcom-block-editor`, remove `@wordpress` dependencies and other dependencies that are externalized by the `dependency-extraction-webpack-plugin` (`lodash`, `jquery`, `react`, ...) from `package.json`. These don't need to be installed in `node_modules`, because they are never accessed from there. Keep installing only the deps that are actually bundled (`classnames`, `debug`). This change also prevents Renovate Bot from submitting nonsensical "pin dependencies" PR like #35216.

This PR also removes `full-site-editing/package-lock.json`: it's not needed -- all locking is done in the root of the monorepo.

In Calypso, upgrade all `@wordpress` dependencies (we actually bundle them there) to the latest working version (we can't upgrade to the very latest ones because of WordPress/gutenberg#17165). And enumerate them all, even the transitive ones, to ensure that we don't get duplicate copies of packages bundled with different versions.

Last but not least, pin the `@wordpress/dependency-extraction-webpack-plugin`. That should make #35216 completely obsolete for now.

**How to test:**
- verify that `full-site-editing` and `wpcom-block-editor` build correctly and work correctly
- verify that Calypso is OK and that the Duplicate Package Checker webpack plugin no longer warns about `@wordpress` packages.

Fixes #35497.
